src/mlpack/methods/ann/rbm/spike_slab_rbm_policy.hpp

```diff
@@ -0,0 +1,223 @@
+   * the visible outputs from the normal distribution with
+   * mean \Lambda^{-1} \sum_{i=1}^N W_i * s_i * h_i and 
+   * variance \Lambda^{-1}
+   * 
+   * @param input consists of spike and slab variables
+   * @param output consits of visible layer.
+   * @param norm norm used for rejection sampling
+   */
+  void SampleVisible(DataType&& input, DataType&& output);
+
+  /**
+   * Sample Hidden function samples 
+   * the slab outputs from the normal distribution with
+   * mean by h_i*\alpha^{-1}*W_i^T*v  and 
+   * variance \alpha&{-1}
+   * 
+   * @param input consists of visible and spike variables
+   * @param output consits of slab units.
+   */
+  void SampleHidden(DataType&& input, DataType&& output);
+
+  // Serialize function
+  template<typename Archive>
+  void Serialize(Archive& ar, const unsigned int /* version */);
+
+  //! Return the initial point for the optimization.
+  const DataType& Parameters() const { return parameter; }
+  //! Modify the initial point for the optimization.
+  DataType& Parameters() { return parameter; }
+
+  //! Get the weight variables
+  arma::cube const& Weight() const { return weight; }
+  arma::cube& Weight() { return weight; }
+
+  //! Get the regulaliser associated with spike variables
+  DataType const& SpikeBias() const { return spikeBias; }
+  DataType& SpikeBias() { return spikeBias; }
+
+  //! Get the regulaliser associated with slab variables
+  ElemType const& SlabPenalty() const { return invSlabPenalty; }
+
+  //! Get the regulaliser associated with visible variables
+  DataType const& VisiblePenalty() const { return visiblePenalty; }
+  DataType& VisiblePenalty() { return visiblePenalty; }
+
+  //! Get the visible size
+  size_t const& VisibleSize() const { return visibleSize; }
+  //! Get the hidden size
+  size_t const& HiddenSize() const { return hiddenSize; }
+  //! Get the pool size
+  size_t const& PoolSize() const { return poolSize; }
+
+ private:
+  /**
+   * Spike Mean function calculates the following distribution
+   * P(h|v) which is given by
+   *  sigm(v^T*W_i*\alpha_i^{-1}*W_i^T*v + b_i)
+   *
+   * @param visible the visible layer
+   * @param spikeMean hidden layer
+   */
+  void SpikeMean(DataType&& visible, DataType&& spikeMean);
+  /**
+   * Sample Spike function samples the spike
+   * function using bernoulli distribution
+   * @param spikeMean indicates P(h|v)
+   * @param spike the sampled 0/1 spike variables
+   */
+  void SampleSpike(DataType&& spikeMean, DataType&& spike);
+
+  /**
+   * SlabMean function calculates the mean of 
+   * normal distribution of P(s|v,h).
+   * Where the mean is given by h_i*\alpha^{-1}*W_i^T*v 
+   * 
+   * @param visible the visible layer units
+   * @param spike the spike variables from hidden layer
+   * @param slabMean the mean of the normal distribution
+   */
+  void SlabMean(DataType&& visible, DataType&& spike,
+      DataType&& slabMean);
+  /**
+   * SampleSlab function calculates the
+   * normal distribution P(s|v,h).
```

Comment:
The function doesn't calculate the distribution. The function samples slab variables.